Softeq.XToolkit.ProjectInfrastructure/ForegroundTaskDeferral.cs

```diff
@@ -59,20 +59,23 @@ public void Complete()
 
             _isInProgress.Set(false);
 
-            _taskCompletition.SetResult(true);
+            _taskCompletition.SetResult(result);
         }
 
-        public void CompleteIfInProgress()
+        public Task<T> WaitForCompletionAsync()
         {
-            if (_isInProgress.Get())
-            {
-                Complete();
-            }
+            return _taskCompletition.Task;
         }
+    }
 
-        public async Task WaitForCompletionAsync()
+    public class ForegroundTaskDeferral : ForegroundTaskDeferral<bool>
```

Comment:
This inheritance actually causes ForegroundTaskDeferral to have both CompleteIfInProgress() and Complete(bool) public methods. I'd say, ForegroundTaskDeferral is not actually a subset of ForegroundTaskDeferral<T>, but both of them are implementations of some abstract class